app/questionnaire/questionnaire_schema.py

```diff
@@ -1109,7 +1113,9 @@ def _get_section_and_block_ids_dependencies_for_progress_source_and_answer_ids_f
                     dependencies_ids_for_progress_value_source["blocks"][section_id] = {
                         identifier: OrderedSet()
```

Comment:
I know this is a merge and this was already there, but I think this should be changed to 
```
dependencies_ids_for_progress_value_source["blocks"][section_id] = {
                        identifier: OrderedSet([current_section_id])
                    }

Reply:
it would fix the problem you're having with `.append` (needs to be `add`)

Reply:
looks like the statement is duplicated, one with passing in the current_section_id and one not, will remove the one without

Reply:
done 👍